core/src/main/scala/org/apache/spark/network/BlockTransferClientSync.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network
+
+import java.nio.ByteBuffer
+
+import scala.concurrent.Promise
+import scala.concurrent.duration.Duration
+
+import org.apache.spark.network.buffer.{FileSegmentManagedBuffer, ManagedBuffer, NioManagedBuffer}
+import org.apache.spark.network.shuffle.{BlockFetchingListener, BlockTransferClient, DownloadFileManager}
+import org.apache.spark.storage.EncryptedManagedBuffer
+import org.apache.spark.util.ThreadUtils
+
+private[spark] class BlockTransferClientSync(val blockTransferClient: BlockTransferClient) {
```

Comment:
I'm not sure I follow the class name. Feel to me like it should be `SyncBlockTransferClient` instead, since the client is synchronous (instead of this being some kind of synchronization object).

Reply:
done